### PR TITLE
Add JmsEventAggregator

### PR DESCRIPTION
**Pull request by meghasemim1999**

### CODE REVIEW

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsConnectionOptions.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    /// <summary>
+    /// Represents a class containing options for configuring a JMS connection.
+    /// </summary>
+    public class JmsConnectionOptions
+    {
+        /// <summary>
+        /// For example: tcp://localhost:61616
+        /// </summary>
+        public string BrokerUri { get; set; }
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsConnectionOptions.cs]: Please use get only

File: Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs [hunk 2]
```diff
@@ -18,11 +18,11 @@ namespace BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregato
     public interface IEvent<T> : IEvent where T : IEventDataModel
     {
         /// <summary>
-        /// Gets the data model associated with the event. The data model serves as a container for structured 
+        /// Gets or sets the data model associated with the event. The data model serves as a container for structured 
         /// information, allowing events to convey specific details or context. By including data models in events, 
         /// components can make use of this information for various purposes, such as processing, logging, or decision-making.
         /// </summary>
-        T DataModel { get; }
+        T DataModel { get; set; }
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs]: Why?

File: Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/ISubscriber.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ public interface ISubscriber
         /// <typeparam name="TEvent">The type of event to subscribe to.</typeparam>
         /// <typeparam name="TEventDataModel">The type of data model associated with the event.</typeparam>
         /// <param name="eventReceiver">The event receiver to subscribe.</param>
-        void Subscribe<TEvent, TEventDataModel>(IEventReceiver eventReceiver) where TEvent : IEvent<TEventDataModel> where TEventDataModel : IEventDataModel;
+        void Subscribe<TEvent, TEventDataModel>(IEventReceiver eventReceiver) where TEvent : IEvent<TEventDataModel>, new() where TEventDataModel : IEventDataModel;
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/ISubscriber.cs]: Why?

File: Source/BSN.Commons/Infrastructure/MessageBroker/RabbitMqEventAggregator/RabbitMqEventAggregator.cs [hunk 4]
```diff
@@ -50,7 +50,7 @@ public class RabbitMqEventAggregator : IEventAggregator
         }
 
         /// <inheritdoc />
-        public void Subscribe<TEvent, TEventDataModel>(IEventReceiver eventReceiver) where TEvent : IEvent<TEventDataModel> where TEventDataModel : IEventDataModel
+        public void Subscribe<TEvent, TEventDataModel>(IEventReceiver eventReceiver) where TEvent : IEvent<TEventDataModel>, new() where TEventDataModel : IEventDataModel
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/RabbitMqEventAggregator/RabbitMqEventAggregator.cs]: Please remove `new()`

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            
+            consumer.Listener += message =>
+            {
+                if (message is ITextMessage textMessage)
+                {
+                    string serializedEvent = textMessage.Text;
+                    
+                    TEventDataModel eventDataModel = JsonConvert.DeserializeObject<TEventDataModel>(serializedEvent);
+                    
+                    TEvent @event = new TEvent
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs]: Discussed with phone ;)

**Review by soroshsabz — CHANGES_REQUESTED** [review]

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsConnectionOptions.cs (hunk 1 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsConnectionOptions.cs]: done

File: Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs (hunk 2 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs]: reverted

File: Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/ISubscriber.cs (hunk 3 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/ISubscriber.cs]: reverted

File: Source/BSN.Commons/Infrastructure/MessageBroker/RabbitMqEventAggregator/RabbitMqEventAggregator.cs (hunk 4 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/RabbitMqEventAggregator/RabbitMqEventAggregator.cs]: done

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs (hunk 5 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs]: done

File: Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs [hunk 6]
```diff
@@ -18,7 +18,7 @@ namespace BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregato
     public interface IEvent<T> : IEvent where T : IEventDataModel
     {
         /// <summary>
-        /// Gets the data model associated with the event. The data model serves as a container for structured 
+        /// Gets or sets the data model associated with the event. The data model serves as a container for structured
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs]: Why change comment?

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        {
+            _subscriptionManager = eventAggregatorSubscriptionManager ?? new InMemoryEventAggregatorSubscriptionManager();
+
+            IConnectionFactory factory = new ConnectionFactory(jmsConnectionOptions.BrokerUri);
+            _connection = factory.CreateConnection();
+            _connection.Start();
+            _session = _connection.CreateSession();
+            _destination = SessionUtil.GetDestination(_session, jmsConnectionOptions.QueueName);
+            
+            _producer = _session.CreateProducer(_destination);
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs]: Why does not use producers?

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        /// <inheritdoc />
+        public void Subscribe<TEvent, TEventDataModel>(IEventReceiver eventReceiver) where TEvent : IEvent<TEventDataModel> where TEventDataModel : IEventDataModel
+        {
+            string eventName = typeof(TEvent).Name;
+            
+            _subscriptionManager.AddSubscription<TEvent>(eventReceiver);
+            
+            var consumer = _session.CreateConsumer(_destination);
```
Comment (soroshsabz) [inline Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs]: What is behavior, if call subscribe with same input twice?

**Review by soroshsabz — CHANGES_REQUESTED** [review]

File: Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs (hunk 6 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/EventContracts/EventAggregator/Contracts/IEvent.cs]: done

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs (hunk 7 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs]: done

File: Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs (hunk 8 above)
Comment (meghasemim1999) [reply Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs]: changed it a bit
now it does not add a new consumer for same eventName
but adds new Listener for new eventRecievers

**Review by soroshsabz — APPROVED** [review]

**Pull request merged**